api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -101,4 +101,8 @@ DbUser syncEraCommonsStatusUsingImpersonation(DbUser user, Agent agent)
 
   // same as the above, but throw NotFoundException if not found
   DbUser getByUsernameOrThrow(String username);
+
+  Optional<DbUser> getByDatabaseId(long databaseId);
+
+  void updateBypassTime(long userDatabaseId, AccessBypassRequest accessBypassRequest);
```

Comment:
Arguably I could split this one out, since it's not needed here, but for simplicity I went one class or file at a time.